Havtorn/Source/Engine/Core/Color.h

```diff
@@ -0,0 +1,111 @@
+// Copyright 2022 Team Havtorn. All Rights Reserved.
+
+#pragma once
+
+namespace Havtorn
+{
+	// Represents a color in RGBA 0-255 range.
+	struct HAVTORN_API SColor
+	{
+		static const F32 FloatMax;
+		static const F32 FloatMin;
+		static const U8 U8Max;
+
+		U8 R, G, B, A = 255;
+
+		static const SColor Red;
+		static const SColor Green;
+		static const SColor Blue;
+		static const SColor Black;
+		static const SColor White;
+		static const SColor Grey;
+		static const SColor Teal;
+		static const SColor Orange;
+		static const SColor Magenta;
+		static const SColor Yellow;
+		// TODO.AG: Add more color presets
+
+		inline SColor();
+		// Alpha is set to 255.
+		inline SColor(U8 monochrome);
+		inline SColor(U8 r, U8 g, U8 b);
+		inline SColor(U8 r, U8 g, U8 b, U8 a);
+		// Expects SVector values to be [0.0f - 1.0f] range.
+		inline SColor(const SVector& rgb);
+		// Expects SVector4 values to be [0.0f - 1.0f] range.
+		inline SColor(const SVector4& rgba);
+
+		// Returns the RGB values in a [0.0f - 1.0f] range.
+		inline SVector AsVector() const;
+		// Returns the RGBA values in a [0.0f - 1.0f] range.
+		inline SVector4 AsVector4() const;
+
+		// Expects SVector values to be [0.0f - 1.0f] range.
+		inline static U8 ToU8Range(const F32 c);
+		// Returs color value in a [0.0f - 1.0f] range.
+		inline static F32 ToFloatRange(const U8 c);
+	};
+
+	SColor::SColor() {}
+
+	SColor::SColor(U8 monochrome)
+		: R(monochrome)
+		, G(monochrome)
+		, B(monochrome)
+	{}
+
+	SColor::SColor(U8 r, U8 g, U8 b)
+		: R(r)
+		, G(g)
+		, B(b)
+	{}
+
+	SColor::SColor(U8 r, U8 g, U8 b, U8 a)
+		: R(r)
+		, G(g)
+		, B(b)
+		, A(a)
+	{}
+
+	SColor::SColor(const SVector& rgb)
+		: R(ToU8Range(rgb.X))
+		, G(ToU8Range(rgb.Y))
+		, B(ToU8Range(rgb.Z))
+	{}
+
+	SColor::SColor(const SVector4& rgba)
+		: R(ToU8Range(rgba.X))
+		, G(ToU8Range(rgba.Y))
+		, B(ToU8Range(rgba.Z))
+		, A(ToU8Range(rgba.W))
+	{}
+
+	SVector SColor::AsVector() const
+	{
+		SVector v;
+		v.X = ToFloatRange(R);
+		v.Y = ToFloatRange(G);
+		v.Z = ToFloatRange(B);
+		return v;
+	}
+
+	SVector4 SColor::AsVector4() const
+	{
+		SVector4 v;
+		v.X = ToFloatRange(R);
+		v.Y = ToFloatRange(G);
+		v.Z = ToFloatRange(B);
+		v.W = ToFloatRange(A);
+		return v;
+	}
+
+	U8 SColor::ToU8Range(const F32 c)
+	{
+		return static_cast<U8>(UMath::Clamp(c, FloatMin, FloatMax) * static_cast<F32>(U8Max));
```

Comment:
since you're always casting U8Max before use, I'd let U8Max be a float (255.0f) and also define U8MaxRcp or U8MaxReciprocal (1.0f / 255.0f) and multiply with that in the next function :)